packages/next/src/build/webpack-config.ts

```diff
@@ -1198,12 +1198,14 @@ export default async function getBaseWebpackConfig(
                   chunks: 'all' as const,
                   name: 'framework',
                   layer: RSPACK_DEFAULT_LAYERS_REGEX,
-                  test: /[/]node_modules[/](react|react-dom|next[/]dist[/]compiled[/](react|react-dom)(-experimental)?)[/]/,
+                  test: new RegExp(
+                    `^(${topLevelFrameworkPaths.map((p) => `(${p.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')})`).join('|')})`
+                  ),
                   priority: 40,
                   enforce: true,
                 },
                 lib: {
-                  test: /[/]node_modules[/](?!.*\.(css|scss|sass|less|styl)$)/,
+                  test: /[\\/]node_modules[\\/](?!.*\.(css|scss|sass|less|styl)$)/,
```

Comment:
It seems a bit weird that webpack/rspack wouldn't normalize paths before passing them to the regex?

Reply:
This is my mistake. I noticed that it is different from other regular expressions, and my first thought was that I wrote it wrong.